### PR TITLE
Possible fix for #1043

### PR DESCRIPTION
**Pull request by enoch85**
Fixes https://github.com/nextcloud/vm/issues/1043

1. Abort if the extraction fails'
2. Check that network is OK before downloading.

Needs serious testing.

### CODE REVIEW

**Comment by szaimen** [comment]
So I have updated a test VM successfull to the new version. But it is not my productive instance and I don't have a slow internet connection. So this might be not representative for slow internet connections and large instances.

File: lib.sh [hunk 1]
```diff
@@ -715,7 +715,13 @@ cd $HTML
 print_text_in_color "$ICyan" "Downloading $STABLEVERSION..."
 curl -fSLO --retry 3 "$NCREPO/$STABLEVERSION.tar.bz2"
 mkdir -p "$GPGDIR"
-curl_to_dir "$NCREPO" "$STABLEVERSION.tar.bz2.asc" "$GPGDIR"
+if network_ok
+then
+    curl_to_dir "$NCREPO" "$STABLEVERSION.tar.bz2.asc" "$GPGDIR"
+else
+    msg_box There seems to be an issue with your network, please try again later.\nThis script will exit."
```
Comment (szaimen) [inline lib.sh]: I think this should be:
`    msg_box "There seems to be an issue with your network, please try again later.\nThis script will exit."`

**Comment by szaimen** [comment]
BTW: I have had an issue there:
if [ "${CURRENTVERSION%%.*}" -ge "15" ]
I think this check doesn't work but I am not completely sure about this.

**Comment by szaimen** [comment]
And isn't there a possibility to directly compare the checksum of the file on the server against the downloaded file? I think this would be more robust... Or is this already implemented?

**Comment by szaimen** [comment]
@enoch85 actually the update got through successfully but there seems something to be really broken with the new sidebar. Don't know if that has to do with the update script.
sometimes the sidear doesnt load at all or if you want to create a share link it doesnt work.

**Comment by szaimen** [comment]
> @enoch85 actually the update got through successfully but there seems something to be really broken with the new sidebar. Don't know if that has to do with the update script.
> sometimes the sidear doesnt load at all or if you want to create a share link it doesnt work.

that was due to my webbrowser. it works now.

**Comment by szaimen** [comment]
The update on my instance was successfull

File: lib.sh (hunk 1 above)
Comment (enoch85) [reply lib.sh]:
```suggestion
    msg_box There seems to be an issue with your network, please try again later.\nThis script will exit."
```

Yes, good catch!

**Comment by enoch85** [comment]
> And isn't there a possibility to directly compare the checksum of the file on the server against the downloaded file? I think this would be more robust... Or is this already implemented?

Yeah that would be the best solution.

I will work on something like that now.

**Comment by enoch85** [comment]
> The update on my productive instance was successfull

Did you use this PRs `curl` command in the top? Otherwise you tested master code.

**Comment by enoch85** [comment]
```
SHA256SUM="$(sha256sum $STABLEVERSION.tar.bz2 | awk '{print $1'})"
SHA256SUM_VERIFY="$(cat $STABLEVERSION.tar.bz2.sha256 | awk '{print $1'})"

if "$SHA256SUM_VERIFY" = "$$SHA256SUM"
then
    msg_box "The checksums of the files didn't match, please try again."
    exit
fi
```

This doesen't work yet, but it's a start.

**Comment by enoch85** [comment]
Please test @szaimen 

Thanks!

**Comment by szaimen** [comment]
@enoch85 so I have tested the new update script and it seems to work. (Updated a test VM successfully from 17 to 18) but initially I had an internal server error after the update and had to manually restart the VM to make it work. Shouldn't be a restart considered after executing the update script?

File: lib.sh [hunk 2]
```diff
@@ -12,6 +12,7 @@ HTML=/var/www
 NCDATA=/mnt/ncdata
 SNAPDIR=/var/snap/spreedme
 GPGDIR=/tmp/gpg
+SHA256_DIR=/tmp/shas56
```
Comment (szaimen) [inline lib.sh]: I am pretty sure, that this should be 
`SHA256_DIR=/tmp/sha256`
But its not an important change.

Comment (enoch85) [reply lib.sh]: Please add it to the "some improvements PR"

**Pull request merged**